### PR TITLE
Duckduckgo

### LINKED ISSUE #7: [FEAT/SE] Duckduckgo
**Describe your feature request**
Create a module for parsing [Duckduckgo](https://duckduckgo.com) search results

**Screenshots**
If applicable, add screenshots to help explain your request.

**Additional context**

### PR DESCRIPTION
**Pull request by aleksasiriski**
Moved to https://github.com/tminaorg/brzaguza/pull/25

### CODE REVIEW

File: src/structures/structures.go [hunk 1]
```diff
@@ -29,14 +29,17 @@ type Options struct {
...
-	Description string
-	Response    *colly.Response
+	URL          string
+	Rank         int
+	SERank       int
+	SEPage       int
+	SEOnPageRank int
+	Title        string
+	Description  string
+	SearchEngine string
```
Comment (aleksasiriski) [inline src/structures/structures.go:41]: Should be array (go slice) of SEs where this result was found.

We should keep values for SERank, SEPage and SEOnPageRank as arrays (go slices) as well.

This can later become the average value between all SEs and used better for ranking instead of relying on the rank from the first SE that returns the result. Since this all happens while still returning results from SEs it won't impact performance (because the latency to SEs is much greater and it can all happen concurrently).

File: src/engines/google/google.go [hunk 2]
```diff
@@ -9,12 +9,13 @@ import (
```
Comment (aleksasiriski) [inline src/engines/google/google.go:9]: It's bad practice to change one feature in another feature's PR (at least mention it in the description or even better TITLE of the PR). Optimal solution would be a separate PR.

File: src/bucket/bucket.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
+package bucket
```
Comment (aleksasiriski) [inline src/bucket/bucket.go:1]: I like this name (:

File: src/engines/duckduckgo/duckduckgo.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+		var titleText string
+		var descText string
+		var rrank int
+
+		var reqBody string = e.Request.Ctx.Get("body")
+		var page int
+		fmt.Sscanf(reqBody, "q="+query+"&dc=%d", &page)
+		page = page/20 + 1
+
+		e.DOM.Children().Each(func(i int, row *goquery.Selection) {
```
Comment (aleksasiriski) [inline src/engines/duckduckgo/duckduckgo.go:101]: Not quite sure what happens here but I tested it and it works.

File: src/structures/structures.go (hunk 1 above)
Comment (k4lizen) [reply src/structures/structures.go:41]: Agreed
